src/_pytest/pathlib.py

```diff
@@ -668,7 +668,7 @@ def _import_module_using_spec(
     parent_module: ModuleType | None = None
     if parent_module_name:
         parent_module = sys.modules.get(parent_module_name)
-        if parent_module is None:
+        if parent_module is None or not hasattr(parent_module, "__path__"):
```

Comment:
Can we have a comment explaining the `or not hasattr` bit? Also probably good to move it to a well named variable.


```suggestion
        # ...
        stdlib_module = not hasattr(parent_module, "__path__")
        if parent_module is None or stdlib_module:
```

While one can use `git blame` to discover that, this is the kind of thing that tends to get buried/hidden as code is moved/changed.

Reply:
Agreed. I suggest using `need_reimport` as the variable name. The key issue is the lack of the `__path__` attribute, regardless of the module's source. If a module with the same name exists in the current working directory and is imported first, the same error can occur.